### PR TITLE
janus-gateway: fix build, 1.2.3 -> 1.2.4

### LINKED ISSUE #350048: Build failure: janus-gateway
### Steps To Reproduce

1. `nix build github:nixos/nixpkgs/9b8bdbfa9da81a5df1249faacc6a9bd5272cecd1#janus-gateway`

### Build log

https://gist.github.com/jackwilsdon/7efe8a2c3b44c8a3844355f434540b32

### Notify maintainers

@fpletz

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.2, NixOS, 24.11 (Vicuna), 24.11.20241006.50b3bd3`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.8`
 - nixpkgs: `/nix/store/ywvkxmphiy3d110p70qj174ici3zmhwg-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by xokdvium**
Fixes https://github.com/NixOS/nixpkgs/issues/350048

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by xokdvium** [comment]
CC @jackwilsdon. This fixes the build, but I'm not sure how to test if the resulting binary works as expected. Could you see if this works for you?

**Comment by jackwilsdon** [comment]
I don't really know how to test it sorry - it's depended upon by `ustreamer` (which I use), but I don't use the `janus-gateway` integration.

I did manage to get it to start a web server using this sample configuration: https://github.com/meetecho/janus-gateway/blob/b092e3d342c997e45dd3918952ea7d638b921e1e/conf/janus.transport.http.jcfg.sample

File: pkgs/servers/janus-gateway/default.nix [hunk 1]
```diff
@@ -44,6 +44,8 @@ stdenv.mkDerivation rec {
 
   makeFlags = [
     "BORINGSSL_LIBS=-L${lib.getLib boringssl}/lib"
+    # Linkding with CXX because boringssl static libraries depend on C++ stdlib.
```
[the remark this answers is not in the document]
Comment (NickCao) [reply pkgs/servers/janus-gateway/default.nix:47]: Exactly.

[the remark this answers is not in the document]
Comment (xokdvium) [reply pkgs/servers/janus-gateway/default.nix:47]: Done

**Review by fpletz — APPROVED** [review]
LGTM. Force-pushed small typo fix for the comment, added reformat, by-name migration and version bump while we are at it. Unfortunately the new version with the included fix mentioned in the upstream issue doesn't fix the build for us.

**Review by fpletz — APPROVED** [review]

**Pull request merged**